packages/eds-core-react/src/components/SideBar/SideBarAccordion/index.tsx

```diff
@@ -0,0 +1,315 @@
+) {
+  const accordionId: string = useId(id, 'accordion')
+  const [menuIsOpen, setMenuIsOpen] = useState<boolean>(false)
+  const [accordionIsOpen, setAccordionIsOpen] = useState<boolean>(false)
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement>(null)
+  const { isOpen } = useSideBar()
+
+  const combinedRefs = useMemo(
+    () => mergeRefs<HTMLButtonElement>(setAnchorEl, ref),
+    [ref],
+  )
+
+  const closeMenu = () => {
+    setMenuIsOpen(false)
+  }
+
+  const onClickWhenSidePanelExpanded = () => {
+    if (toggleExpand === undefined) {
+      setAccordionIsOpen(!accordionIsOpen)
+    } else {
+      toggleExpand()
+    }
+    onClick && onClick()
+  }
+
+  const onClickWhenSidePanelClosed = () => {
+    setMenuIsOpen(!menuIsOpen)
+    onClick && onClick()
+  }
+
+  const getIconColor = () => {
+    return active ? typographyActiveColor : typographyColor
+  }
+
+  const showPanel = toggleExpand !== undefined ? isExpanded : accordionIsOpen
+
+  if (isOpen) {
+    return (
+      <MenuItem $active={active} $isExpanded={showPanel}>
+        <AccordionHeader>
+          <OpenSidebarButton
+            ref={ref}
+            id={`header_${accordionId}`}
+            aria-expanded={showPanel}
+            aria-controls={`panel_${accordionId}`}
+            onClick={onClickWhenSidePanelExpanded}
+            {...rest}
+          >
+            {icon && <Icon data={icon} color={getIconColor()} />}
+            <ItemText variant="cell_text" group="table" $active={active}>
+              {label}
+            </ItemText>
+            <Icon
+              data={showPanel ? chevron_up : chevron_down}
+              size={24}
+              color={typographyColor}
+            />
+          </OpenSidebarButton>
+        </AccordionHeader>
+        {showPanel && (
+          <Panel
+            id={`panel_${accordionId}`}
+            role="region"
+            aria-labelledby={`header_${accordionId}`}
+          >
+            {children}
+          </Panel>
+        )}
+      </MenuItem>
+    )
+  }
+
+  const tooltip = menuIsOpen ? '' : label
+
+  return (
+    <>
+      <Tooltip title={tooltip} placement="right">
+        <MenuItem $active={active} $isExpanded={showPanel}>
+          <AccordionHeader>
+            <ClosedSidebarButton
+              ref={combinedRefs}
+              id="anchor-default"
+              aria-haspopup="true"
+              aria-expanded={menuIsOpen}
+              aria-controls="menu-default"
+              onClick={onClickWhenSidePanelClosed}
+              {...rest}
+            >
+              {icon && <Icon data={icon} color={getIconColor()} />}
+              <AccordionIcon
+                data={arrow_drop_down}
+                size={24}
+                color={getIconColor()}
+              />
+            </ClosedSidebarButton>
+          </AccordionHeader>
+        </MenuItem>
+      </Tooltip>
+      <Menu
+        open={menuIsOpen}
+        placement={'right-start'}
+        id="menu-default"
```

Comment:
This should probably not have an id? otherwise there could be several divs with the same id